### DISCUSSION TITLE
Retrieve response from Home Assistant system_bridge.send_command?

### DISCUSSION THREAD

**Discussion by backslashV** [timmo001/system-bridge#556] [Help] [answered]
Hi,
How do I get the response from system_bridge.send_command? I need to execute "powercfg -lastwake" and get the response.

**Comment by timmo001** [comment, marked as answer]
If you are asking about the Home Assistant integration, the command is currently a send and forget. It currently has no functionality to handle the response.

This is possible though, using the API manually:

https://system-bridge.timmo.dev/docs/api?endpoint=command


---

There is now an open PR to return the response as a service response https://github.com/home-assistant/core/pull/100055/files

---

**Reply by backslashV** [reply]
Thank you for your response. `curl -X POST http://192.168.0.100:9170/ -d '{ "command": "powercfg", "arguments": "-lastwake", "api_key": "actual_api" }'` returns
`{"statusCode":403,"message":"Forbidden resource","error":"Forbidden"}`
What am I doing wrong?

**Reply by timmo001** [reply]
`api-key` should be part of the header

**Reply by timmo001** [reply]
Your URL should also be `http://hostname:9170/command`

**Reply by backslashV** [reply]
Thanks. `curl -H '{"api_key": "actual_api"}' -d '{"command": "powercfg", "arguments": "-lastwake"}' -X POST http://192.168.0.100:9170/command`

seemingly executes without any errors but doesn't print the results.

**Reply by backslashV** [reply]
If what I have is still incorrectly formed, please post the correct one.

**Reply by backslashV** [reply]
Part of the issue is resolved. I was using api_key instead of api-key. Using the latter, I am getting 201 Created, but don't receive the output in the response section.

**Reply by timmo001** [reply]
I probably should update the docs with this example, but the allowed parameters can be found here:

https://github.com/timmo001/system-bridge/blob/master/src/command/dto/create-command.dto.ts

add:

```json
"wait": true 
```

to wait for the response, otherwise your command will be async

**Reply by timmo001** [reply]
`"message": "foo"` will be in the response with the stdout

**Reply by backslashV** [reply]
Thanks a lot. That worked. Yes, I too believe the documents should be updated.

**Reply by aniceberg** [reply]
Has this functionality been removed? I would like to send terminal commands to restart applications running on an RPi host and at most it seems I can send keyboard presses to the foreground application.
